File: Easy/Are The Clumps Normal/Are The Clumps Normal.lua
local function count_clumps(number, base_value)
    local clumps = 0
    local prev_remainder = nil

    for digit in number:gmatch(".") do
        local remainder = tonumber(digit) % base_value

        if prev_remainder == nil or remainder ~= prev_remainder then
            clumps = clumps + 1
            prev_remainder = remainder
        end
    end

    return clumps
end

local function find_first_deviation(number)
    local previous_clumps = nil

    for base = 2, 9 do
        local clumps = count_clumps(number, base)

        if previous_clumps ~= nil and clumps < previous_clumps then
            return base
        end

        previous_clumps = clumps
    end

    return "Normal"
end

local N = io.read("*l"):match("^%s*(.-)%s*$")
print(find_first_deviation(N))

File: Easy/Are The Clumps Normal/Are The Clumps Normal.cs
using System;

class Program
{
    static int CountClumps(string number, int baseValue)
    {
        int clumps = 0;
        int? prevRemainder = null;

        foreach (char digit in number)
        {
            int remainder = (digit - '0') % baseValue;

            if (prevRemainder == null || remainder != prevRemainder)
            {
                clumps++;
                prevRemainder = remainder;
            }
        }

        return clumps;
    }

    static object FindFirstDeviation(string N)
    {
        int? previousClumps = null;

        for (int B = 2; B <= 9; B++)
        {
            int clumps = CountClumps(N, B);

            if (previousClumps != null && clumps < previousClumps)
            {
                return B;
            }

            previousClumps = clumps;
        }

        return "Normal";
    }

    static void Main()
    {
        string N = Console.ReadLine().Trim();
        Console.WriteLine(FindFirstDeviation(N));
    }
}
